### PR TITLE
Adding HDemucs pipeline to source_separation pipeline. 

### PR DESCRIPTION
**Pull request by skim0514**
Adding HDemucs new pipeline to source_separation pipeline. Add integration testing.

### CODE REVIEW

**Comment by nateanl** [comment]
build_doc job is down. Could you post the documentation built locally?

File: torchaudio/prototype/pipelines/source_separation_pipeline.py [hunk 1]
```diff
@@ -64,9 +64,23 @@ def get_model(self) -> torch.nn.Module:
...
+    _sample_rate=44100,
+)
+
 CONVTASNET_BASE_LIBRI2MIX.__doc__ = """Pre-trained ConvTasNet pipeline for source separation.
     The underlying model is constructed by :py:func:`torchaudio.prototyoe.models.conv_tasnet_base`
     and utilizes weights trained on Libri2Mix using training script ``lightning_train.py``
     `here <https://github.com/pytorch/audio/tree/main/examples/source_separation/>`__ with default arguments.
     Please refer to :py:class:`SourceSeparationBundle` for usage instructions.
     """
+
```
Comment (nateanl) [inline torchaudio/prototype/pipelines/source_separation_pipeline.py:80]: The doc should be next to the bundle class. Could you move `HDEMUCS_HIGH_MUSDB_PLUS` here?

File: torchaudio/prototype/pipelines/source_separation_pipeline.py [hunk 2]
```diff
@@ -64,9 +64,23 @@ def get_model(self) -> torch.nn.Module:
...
+HDEMUCS_HIGH_MUSDB_PLUS.__doc__ = """Pre-trained Hybrid Demucs pipeline for music source separation. 
+    [:footcite:`defossez2021hybrid`] The underlying model is constructed by
```
Comment (nateanl) [inline torchaudio/prototype/pipelines/source_separation_pipeline.py:82]:
```suggestion
HDEMUCS_HIGH_MUSDB_PLUS.__doc__ = """Pre-trained *Hybrid Demucs* [:footcite:`defossez2021hybrid`] pipeline
    for music source separation. The underlying model is constructed by 
```

File: torchaudio/prototype/pipelines/source_separation_pipeline.py [hunk 3]
```diff
@@ -64,9 +64,23 @@ def get_model(self) -> torch.nn.Module:
...
     The underlying model is constructed by :py:func:`torchaudio.prototyoe.models.conv_tasnet_base`
     and utilizes weights trained on Libri2Mix using training script ``lightning_train.py``
     `here <https://github.com/pytorch/audio/tree/main/examples/source_separation/>`__ with default arguments.
     Please refer to :py:class:`SourceSeparationBundle` for usage instructions.
     """
+
+HDEMUCS_HIGH_MUSDB_PLUS.__doc__ = """Pre-trained Hybrid Demucs pipeline for music source separation. 
+    [:footcite:`defossez2021hybrid`] The underlying model is constructed by 
+    :py:func:`torchaudio.prototyoe.models.hdemucs_high` and utilizes weights trained on *MUSDB-HQ* 
+    [:footcite:`MUSDB18HQ`] as well as other further sources. Training was performed on original HDemucs repo
```
Comment (nateanl) [inline torchaudio/prototype/pipelines/source_separation_pipeline.py:84]: It's better to mention the sample rate of the training data.

Comment (nateanl) [reply torchaudio/prototype/pipelines/source_separation_pipeline.py:84]: I'm not sure what are the other further sources. Can you explicitly describe it?

Comment (nateanl) [inline torchaudio/prototype/pipelines/source_separation_pipeline.py:84]: there is a whitespace in the end, please also remove that.
```suggestion
    [:footcite:`MUSDB18HQ`] as well as other further sources. Training was performed on original HDemucs repository
```

**Review by carolineechen — COMMENTED** [review]
can you rebase onto main for the updated circle ci tests?

File: test/integration_tests/source_separation_pipeline_test.py [hunk 4]
```diff
@@ -1,16 +1,24 @@
...
+@pytest.mark.parametrize(
+    "bundle,task,channel,expected_score",
+    [
+        [CONVTASNET_BASE_LIBRI2MIX, "speech_separation", 1, 8.1373],
+        [HDEMUCS_HIGH_MUSDB_PLUS, "music_separation", 2, 8.7480],
+    ],
+)
+def test_source_separation_models(bundle, task, channel, expected_score, mixture_source, clean_sources):
     """Integration test for the source separation pipeline.
     Given the mixture waveform with dimensions `(batch, 1, time)`, the pre-trained pipeline generates
```
Comment (carolineechen) [inline test/integration_tests/source_separation_pipeline_test.py:23]:
```suggestion
    Given the mixture waveform with dimensions `(batch, channels, time)`, the pre-trained pipeline generates
```

File: test/integration_tests/source_separation_pipeline_test.py [hunk 5]
```diff
@@ -1,16 +1,24 @@
...
+from torchaudio.prototype.pipelines import CONVTASNET_BASE_LIBRI2MIX, HDEMUCS_HIGH_MUSDB_PLUS
 
 
 sys.path.append(os.path.join(os.path.dirname(__file__), "..", "..", "examples"))
 from source_separation.utils.metrics import PIT, sdr
 
 
-def test_source_separation_models(mixture_source, clean_sources):
+@pytest.mark.parametrize(
+    "bundle,task,channel,expected_score",
```
Comment (carolineechen) [inline test/integration_tests/source_separation_pipeline_test.py:15]: nit: channels

Comment (carolineechen) [reply test/integration_tests/source_separation_pipeline_test.py:15]: nvm, this is following convention from https://github.com/pytorch/audio/blob/main/CONTRIBUTING.md

**Pull request closed without merging**